File: 71-simplify-path/71-simplify-path.java
class Solution {
    public String simplifyPath(String path) {
        String[] arr=path.split("/");
        Stack<String> s=new Stack();
        for(String str: arr){
            if(str.equals(".") || str.length()==0){
                continue;
            }else if(str.equals("..")){
                if(s.size()>0){
                s.pop();
                }
            }else{
                s.push(str);
            }
        }
        String ans="";
        while(s.size()>0){
            ans="/"+s.pop()+ans;
        }
        if(ans.length()==0){
            ans="/";
        }
        return ans;
    }
}

File: 71-simplify-path/71-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        path = path.split('/')
        print(path)
        stack = []
        for c in path:
            if c:
                if c == '.':
                    continue
                elif c == '..':
                    if stack:
                        stack.pop()
                    else:
                        continue
                else:
                    stack.append(c)
        
        
        return '/'+'/'.join(stack)